bot.py

```diff
@@ -67,6 +67,7 @@
             # Start a git pull to update bot
             print(str(subprocess.Popen('git pull', shell=True, stdout=subprocess.PIPE).stdout.read()))
             await client.send_message(message.channel, 'Update Successful! Restarting...')
+            subprocess.Popen('echo ' + message.channel + ' | cat > LastId', shell=True, stdout=subprocess.PIPE).stdout.read()
```

Comment:
Adrian: I know you like bash (I do too), and I understand it's easier to do it in this way, but you should avoid using subprocesses. There are very, _very_ few occasions where using a bash command is the best way to do things. It's best to look up a native Python (or your current language) method. This way, your application will be _significantly_ faster, lighter, and cross-platform.

(And yes I know you corrected it here. Just a tip for the future.)